src/function/aggregate/distributive/approx_count.cpp

```diff
@@ -86,10 +86,8 @@ struct ApproxCountDistinctFunctionString : ApproxCountDistinctFunctionBase {
 		if (nullmask[idx]) {
 			return;
 		}
-		string value = input[idx].GetString();
-		//! FIXME: Hash this in the future
-		auto size = value.size() > 8 ? 8 : value.size();
-		state->log->Add((uint8_t *)&value, size);
+		auto str_hash = std::hash<std::string> {}(input[idx].GetString());
```

Comment:
Can you use our own hash instead of std::hash? This is not very efficient as it constructs a std::string object which involves memory allocation and copying the string.